### PR TITLE
JobQueue

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/libsync/jobqueue.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+class OWNCLOUDSYNC_EXPORT JobQueue : public QObject
+{
+    Q_OBJECT
+public:
+    JobQueue();
+
+    void block(bool block);
+    bool isBlocked() const;
+    bool retry(AbstractNetworkJob *job);
+    bool queue(AbstractNetworkJob *job);
```
Comment (TheOneRing) [inline src/libsync/jobqueue.h:33]: enqueue

File: src/libsync/jobqueue.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+#include <QObject>
+
+namespace OCC {
+
+class AbstractNetworkJob;
+
+class OWNCLOUDSYNC_EXPORT JobQueue : public QObject
+{
+    Q_OBJECT
```
Comment (TheOneRing) [inline src/libsync/jobqueue.h:26]: no need for qobject

**Comment by TheOneRing** [comment]
Doesn't work yet...

File: src/libsync/jobqueue.cpp [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    }
+}
+
+size_t JobQueue::size() const
+{
+    return _jobs.size();
+}
+
+bool JobQueue::needsRetry(AbstractNetworkJob *job) const
+{
```
Comment (dragotin) [inline src/libsync/jobqueue.cpp:106]: I was wondering why this method is in `JobQueue` and not in the `Job` class directly. The job should be the source if retry is needed or not. 

And the implementation indeed only uses job data to find out. So this method could be moved to `Job::needsRetry()`

**Review by dragotin — APPROVED** [review]
Apart from the one comment I made (which is minor) I can not see anything suspicious.

Very good work. Makes the code more clear and solves problems :1st_place_medal: 

However, this is a significant change that needs good testing.

File: src/libsync/jobqueue.cpp (hunk 3 above)
Comment (TheOneRing) [reply src/libsync/jobqueue.cpp]: good point thx

**Pull request merged**